cmd/traffic_cache_tool/CacheScan.cc

```diff
@@ -0,0 +1,335 @@
+/** @file
+
+  A brief file description
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#include "CacheScan.h"
+#include "../../proxy/hdrs/HTTP.h"
+#include "../../proxy/hdrs/HdrHeap.h"
+#include "../../proxy/hdrs/MIME.h"
+#include "../../proxy/hdrs/URL.h"
+
+// using namespace ct;
+
+const int HTTP_ALT_MARSHAL_SIZE = ROUND(sizeof(HTTPCacheAlt), HDR_PTR_SIZE);
+
+namespace ct
+{
+URL_map CacheScan::global_url_map;
+Errata
+CacheScan::Scan()
+{
+  Errata zret;
+  for (int s = 0; s < this->stripe->_segments; s++) {
+    for (int b = 0; b < this->stripe->_buckets; b++) {
+      CacheDirEntry *seg = this->stripe->dir_segment(s);
+      CacheDirEntry *e   = dir_bucket(b, seg);
+      if (dir_offset(e)) {
+        do {
+          // std::cout << "found stuffs here in " << this->stripe->hashText << std::endl;
+          char *stripe_buff2 = (char *)ats_memalign(ats_pagesize(), dir_approx_size(e));
```

Comment:
it gets freed at the end of the function

Reply:
I'm not seeing that - which line does the free? Also, doesn't this get allocated every loop, in which case a single free at the end won't clean them all up.

Reply:
ah sorry I was thinking this function is Stripe::dir_probe() in CacheDefs.cc , that function allocates stripe_buff2 too. Anyway, I am using this memory for Doc instances. Also this is getting allocated only when there is a doc to read, since offset will be non zero. I don't think we need this doc later, and we are not saving the hdr objects anymore, so I guess I can remove this